### PR TITLE
Migrate to CircleCI 2.0

### PR DESCRIPTION
**Pull request by shakuzen**
Convert to the CircleCI 2.0 format. Switches from Zulu to CircleCI's OpenJDK Docker image.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          key: offline-dependencies-{{ checksum "pom.xml" }}
+          paths:
+            - ~/.m2
+            - zipkin-ui/node_modules
+
+      - run:
+          name: Tests
+          command: |
+            # skip license on circleci due to #1512
+            ./mvnw -Dtest="*Test" -DfailIfNoTests=false test -Dlicense.skip=true
```
Comment (shakuzen) [inline .circleci/config.yml:48]: I inlined what it appeared the `test.sh` script was effectively doing. Let me know if I've missed something or you would prefer to call the `test.sh` file.

File: circle.yml [hunk 2]
```diff
@@ -1,54 +0,0 @@
...
-    - sudo apt-get install xsltproc
-    - ./circleci/go-offline.sh
-    - ./mvnw frontend:install-node-and-npm frontend:npm -pl zipkin-ui
-  cache_directories:
-    - "~/.npm"
-    - "zipkin-ui/node_modules"
-
-database:
-  override:
-    - mysql -u ubuntu -e 'SET GLOBAL innodb_file_format=Barracuda'
```
Comment (shakuzen) [inline circle.yml:38]: I got rid of these as they did not seem needed given the note at the top about MySQL being tested on Travis.

File: circle.yml [hunk 3]
```diff
@@ -1,54 +0,0 @@
...
-    - mysql -u ubuntu circle_test < zipkin-storage/mysql-v1/src/main/resources/mysql.sql
-
-test:
-  override:
-    - ./circleci/test.sh:
-        parallel: true
```
Comment (shakuzen) [inline circle.yml:44]: I'm not sure how this is passed or what it does.

File: .circleci/config.yml (hunk 1 above)
Comment (codefromthecrypt) [reply .circleci/config.yml:48]: will defer to @abesto for any post-merge fixies

**Comment by codefromthecrypt** [comment]
thank you! looks like 2 minutes quicker

**Comment by abesto** [comment]
Also much cleaner. Nice job.

**Pull request merged**